compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -289,9 +289,8 @@ where
         let shard = state.shards.get_shard_by_value(&self.key);
         let job = {
             let mut shard = shard.lock();
-            let job = match shard.active.remove(&self.key).unwrap() {
-                QueryResult::Started(job) => job,
-                QueryResult::Poisoned => panic!(),
+            let QueryResult::Started(job) = shard.active.remove(&self.key).unwrap() else {
```

Comment:
`QueryResult` is not a well-known enum, I'd rather keep the match to emphasize that we panic on `Poisoned` queries.

Reply:
Oh I've missed it. Thanks for pointing it out!